tick/ast/node.go

```diff
@@ -981,6 +1024,101 @@ func newProgram(p position) *ProgramNode {
 	}
 }
 
+func NewProgramNodeFromTickscript(tickscript string) (*ProgramNode, error) {
```

Comment:
I don't really like this function name, but I get that it is a common need to be able to type assert the ast.Node to its concrete type. Should we just have the `Parse` method return the concrete type?